### PR TITLE
Add warnings for bloom filter settings on PairSCollFunc

### LINKED ISSUE #2040: Bloom Filter and other side input size may exceed default cache
`sparseIntersectByKey` and a few other transforms that rely on Bloom Filter or side input may create side input greater than the default 100MB cache. Should we warn this, automatically bump cache, or something else? However cache size is not exposed to us and runner specific.

What about:
- warn if SI size estimate is > 100MB
- document size consideration

That's probably the best we can do without involving runner specific logic.

Optimal BF setting is computed here:
https://github.com/spotify/scio/blob/master/scio-core/src/main/scala/com/spotify/scio/values/PairSCollectionFunctions.scala#L38

### PR DESCRIPTION
**Pull request by anish749**
Fixes #2040

### CODE REVIEW

File: scio-core/src/main/scala/com/spotify/scio/values/PairSCollectionFunctions.scala [hunk 1]
```diff
@@ -59,10 +59,39 @@ private object PairSCollectionFunctions {
...
 
     val totalBytes = width.toLong * numBFs / 8
-    logger.info(s"Optimal Bloom Filter settings for numEntries = $numEntries, fpProb = $fpProb")
-    logger.info(
-      s"BF width = $width, capacity = $capacity, numBFs = $numBFs, total bytes = $totalBytes"
-    )
+    val totalSizeMb = totalBytes / 1024 / 1024
+
+    /*
+     * For sparse transforms, BloomFilters are stored as SideInputs. For some runners, this side
```
Comment (regadas) [inline scio-core/src/main/scala/com/spotify/scio/values/PairSCollectionFunctions.scala]: @anish749 move this up to this method signature? I think you can also move the comments around the runners limits to it. wdyt?

File: scio-core/src/main/scala/com/spotify/scio/values/PairSCollectionFunctions.scala [hunk 2]
```diff
@@ -591,7 +620,7 @@ class PairSCollectionFunctions[K, V](val self: SCollection[(K, V)]) {
     thisNumKeys: Long,
     fpProb: Double
   )(implicit hash: Hash128[K], koder: Coder[K], voder: Coder[V]): Seq[SideInput[MutableBF[K]]] = {
-    val bfSettings = PairSCollectionFunctions.optimalBFSettings(thisNumKeys, fpProb)
+    val bfSettings = PairSCollectionFunctions.optimalBFSettings(self.tfName, thisNumKeys, fpProb)
```
[the remark this answers is not in the document]
Comment (anish749) [reply scio-core/src/main/scala/com/spotify/scio/values/PairSCollectionFunctions.scala]: `tfName` is a property of the `SCollection` and this function is on `PairSCollectionFunctions`.
Where should we pick the right name from? I didn't try running this from a different project.

File: scio-core/src/main/scala/com/spotify/scio/values/PairSCollectionFunctions.scala (hunk 1 above)
Comment (anish749) [reply scio-core/src/main/scala/com/spotify/scio/values/PairSCollectionFunctions.scala]: Yes, this sounds great. I'll changed this.

File: scio-core/src/main/scala/com/spotify/scio/values/PairSCollectionFunctions.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (regadas) [reply scio-core/src/main/scala/com/spotify/scio/values/PairSCollectionFunctions.scala]: @anish749 sorry yeah this is the `SCollection.tfName` somehow i thought it was local val

**Comment by regadas** [comment]
@anish749 Thanks!

**Pull request merged**